enhancements/sig-architecture/34-work-executor-group/README.md

```diff
@@ -0,0 +1,219 @@
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/website/)
+
+## Summary
+
+The proposal adds a new field `executorSubject` to the spec of `ManifestWork` 
+api, to support dynamic identity impersonation upon the execution of 
+`ManifestWork` resources. If the `executorSubject` is absent, it means no
+impersonation will be performed when the work agent applies resources to the
+spoke cluster which is also for the backward compatibility.
+
+## Motivation
+
+### Auditing the API request history from work agent
+
+Currently all the API requests raised by work agent in the managed cluster are
+from a fixed service account ("open-cluster-management-agent/klusterlet-work-sa"
+under the default installation). However in some cases, we need to clarify the
+source/owner identity of the `ManifestWork` resource so that we can record the
+api audits in a finer granularity. This will be helpful for the managed 
+cluster's admin to be capable of knowing "who is operating that `ManifestWork`
+from the hub cluster".
+
+### Separate the identity and its held permission for the work agent
+
+There can be different actors using the work api to distribute resource from
+the hub cluster. Currently the work agent has the cluster-admin access to the
+managed cluster which means that every hub users with write access to the
+work resources will be able to dispatch any resources to the managed cluster.
+This can be a security hole for OCM, e.g. one can apply/overwrite some
+cluster-scoped configuration to the managed cluster which potentially break/halt
+the cluster. Hence, by leverage the "Impersonation" mechanism in Kubernetes,
+we can dynamically switch the identity from work agent's original service
+account, and bind different scopes of RBAC permission to the impersonated 
+identities.
+
+### Goals
+
+- `ManifestWork` resource be able to clarify the target identity to impersonate
+  before writing APIs in the managed cluster.
+
+### Non-Goals
+
+- This proposal doesn't cover which identity to impersonate in the `ManifestWork`
+  api, i.e. it differs depending on use-cases.
+  
+## Proposal
+
+### API Changes
+
+This proposal adds a new field `executorSubject` to the `ManifestWork` api, 
+which is defaulted to nil if absent. A nil executor subject indicates that the 
+work agent doesn't perform any impersonation before requesting the hosting 
+managed cluster.
+
+The content under `executorSubject` is inspired by the original "Subject" 
+struct from Kubernetes' RBAC api:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata: ...
+subjects:
+- kind: User
+  name: dave
+```
+
+A sample of the new `executorSubject` field in Yaml is shown below:
+
+```yaml
+apiVersion: work.open-cluster-management.io/v1
+kind: ManifestWork
+metadata: ...
+spec:
+  executorSubject:
+    type: [ServiceAccount|User] # union discriminator
```

Comment:
I recommend against adding user, but discriminator makes sense to keep.  I could imagine a future with a `SpokeWorkRole` for instance.

Reply:
yes in the executor subject we're referencing a "role" instead of a "user". i removed "User" type and keep only "ServiceAccount".